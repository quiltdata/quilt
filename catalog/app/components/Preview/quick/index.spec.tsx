import * as React from 'react'
import renderer from 'react-test-renderer'

import type * as FileEditor from 'components/FileEditor'

import { QuickPreview, isQuickPreviewAvailable } from './index'

jest.mock('./Markdown', () => ({
  ...jest.requireActual('./Markdown'),
  Render: () => <h1>This is Markdown quick preview</h1>,
}))

describe('app/components/Preview/quick/index.spec.tsx', () => {
  describe('isQuickPreviewAvailable', () => {
    it('should say if quick preview is available', () => {
      const types: FileEditor.EditorInputType[] = [
        'markdown' as const,
        'json' as const,
        null,
      ].map((brace) => ({ brace }))
      expect(types.map(isQuickPreviewAvailable)).toEqual([true, false, false])
    })
  })

  describe('QuickPreview', () => {
    const handle = {
      bucket: 'foo',
      key: 'bar',
    }

    it('renders no value', () => {
      const tree = renderer
        .create(
          <QuickPreview
            {...{ handle, type: { brace: 'markdown' as const }, value: '' }}
          />,
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('renders no preview if unsupported file type', () => {
      const tree = renderer
        .create(
          <QuickPreview {...{ handle, type: { brace: 'json' as const }, value: '' }} />,
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('renders Markdown', () => {
      const tree = renderer
        .create(
          <QuickPreview
            {...{ handle, type: { brace: 'markdown' as const }, value: '=== Title' }}
          />,
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
