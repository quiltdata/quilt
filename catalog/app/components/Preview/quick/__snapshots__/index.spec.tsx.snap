// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app/components/Preview/quick/index.spec.tsx QuickPreview renders Markdown 1`] = `
<h1>
  This is Markdown quick preview
</h1>
`;

exports[`app/components/Preview/quick/index.spec.tsx QuickPreview renders no preview if unsupported file type 1`] = `
<p
  className="MuiTypography-root MuiTypography-body1"
>
  There is no content for quick preview
</p>
`;

exports[`app/components/Preview/quick/index.spec.tsx QuickPreview renders no value 1`] = `
<p
  className="MuiTypography-root MuiTypography-body1"
>
  There is no content for quick preview
</p>
`;
